source/slang/slang-emit-spirv.cpp

```diff
@@ -5406,7 +5406,7 @@ struct SPIRVEmitContext : public SourceEmitterBase, public SPIRVEmitSharedContex
                 }
                 break;
             }
-        case kIROp_DownstreamModuleExportDecoration:
+        case kIROp_HLSLExportDecoration:
```

Comment:
I wonder if the test failures are related to swapping `kIROp_DownstreamModuleExportDecoration` for `kIROp_HLSLExportDecoration` (+ the other similar changes). Perhaps we could have both here?

Reply:
You're right. Maybe it's better if we have both here.

Sorry, I don't quite understand `export` and `public`. Could I assume that `public` is only used in slang-module if it's as function decoration?
As for slang-module, we don't use `export`, downstream ir is incorrect when emitting Spir-V.
As for normal shader (directly emit Spir-V), we don't use `public`, due to slang remove all non-export function.